code/modules/ninja/suit/suit.dm

```diff
@@ -96,59 +107,108 @@ Contents:
 
 	user.adjust_bodytemperature(BODYTEMP_NORMAL - user.bodytemperature)
 
-//Simply deletes all the attachments and self, killing all related procs.
-/obj/item/clothing/suit/space/space_ninja/proc/terminate()
-	qdel(n_hood)
-	qdel(n_gloves)
-	qdel(n_shoes)
-	qdel(src)
-
-
-//Randomizes suit parameters.
-/obj/item/clothing/suit/space/space_ninja/proc/randomize_param()
-	s_cost = rand(1,20)
-	s_acost = rand(20,100)
-	s_delay = rand(10,100)
-	s_bombs = rand(5,20)
-	a_boost = rand(1,7)
-
+/obj/item/clothing/suit/space/space_ninja/get_cell()
+	return cell
 
-//This proc prevents the suit from being taken off.
-/obj/item/clothing/suit/space/space_ninja/proc/lock_suit(mob/living/carbon/human/H)
-	if(!istype(H))
+/obj/item/clothing/suit/space/space_ninja/ui_action_click(mob/user, action)
+	if(istype(action, /datum/action/item_action/initialize_ninja_suit))
+		toggle_on_off()
+		return TRUE
+	if(!s_initialized)
+		to_chat(user, "<span class='warning'><b>ERROR</b>: suit offline. Please activate suit.</span>")
+		return FALSE
+	if(s_coold != 0)
+		to_chat(user, "<span class='warning'><b>ERROR</b>: suit is on cooldown.</span>")
+		return FALSE
+	if(istype(action, /datum/action/item_action/ninjastatus))
+		ninjastatus()
+		return TRUE
+	if(istype(action, /datum/action/item_action/ninjaboost))
+		ninjaboost()
+		return TRUE
+	if(istype(action, /datum/action/item_action/ninjapulse))
+		ninjapulse()
+		return TRUE
```

Comment:
Do a redefined proc instead of a long istype check chain.

Reply:
How exactly am I to do a redefined proc chain for these?  They don't do anything but alert the suit when one of them has been pressed, and share no common proc as far as I'm aware.